test/e2e/autoscaler.go

```diff
@@ -145,21 +156,27 @@ func AutoscalerSpec(ctx context.Context, inputGetter func() AutoscalerSpecInput)
 		mdNodeGroupMaxSize, ok := mdTopology.Metadata.Annotations[clusterv1.AutoscalerMaxSizeAnnotation]
 		Expect(ok).To(BeTrue(), "MachineDeploymentTopology %s does not have the %q autoscaler annotation", mdTopology.Name, clusterv1.AutoscalerMaxSizeAnnotation)
 
-		// Ensure the MachinePoolTopology does NOT have the autoscaler annotations so we can test MachineDeployments first.
-		mpTopology := clusterResources.Cluster.Spec.Topology.Workers.MachinePools[0]
-		if mpTopology.Metadata.Annotations != nil {
-			_, ok = mpTopology.Metadata.Annotations[clusterv1.AutoscalerMinSizeAnnotation]
-			Expect(ok).To(BeFalse(), "MachinePoolTopology %s does have the %q autoscaler annotation", mpTopology.Name, clusterv1.AutoscalerMinSizeAnnotation)
-			_, ok = mpTopology.Metadata.Annotations[clusterv1.AutoscalerMaxSizeAnnotation]
-			Expect(ok).To(BeFalse(), "MachinePoolTopology %s does have the %q autoscaler annotation", mpTopology.Name, clusterv1.AutoscalerMaxSizeAnnotation)
+		if hasMachinePool {
+			// Ensure the MachinePoolTopology does NOT have the autoscaler annotations so we can test MachineDeployments first.
+			mpTopology := clusterResources.Cluster.Spec.Topology.Workers.MachinePools[0]
+			if mpTopology.Metadata.Annotations != nil {
+				_, ok = mpTopology.Metadata.Annotations[clusterv1.AutoscalerMinSizeAnnotation]
+				Expect(ok).To(BeFalse(), "MachinePoolTopology %s does have the %q autoscaler annotation", mpTopology.Name, clusterv1.AutoscalerMinSizeAnnotation)
+				_, ok = mpTopology.Metadata.Annotations[clusterv1.AutoscalerMaxSizeAnnotation]
+				Expect(ok).To(BeFalse(), "MachinePoolTopology %s does have the %q autoscaler annotation", mpTopology.Name, clusterv1.AutoscalerMaxSizeAnnotation)
+			}
 		}
 
 		// Get a ClusterProxy so we can interact with the workload cluster
 		workloadClusterProxy := input.BootstrapClusterProxy.GetWorkloadCluster(ctx, clusterResources.Cluster.Namespace, clusterResources.Cluster.Name)
 		mdOriginalReplicas := *clusterResources.MachineDeployments[0].Spec.Replicas
 		Expect(strconv.Itoa(int(mdOriginalReplicas))).To(Equal(mdNodeGroupMinSize), "MachineDeployment should have replicas as defined in %s", clusterv1.AutoscalerMinSizeAnnotation)
-		mpOriginalReplicas := *clusterResources.MachinePools[0].Spec.Replicas
-		Expect(int(mpOriginalReplicas)).To(Equal(1), "MachinePool should default to 1 replica via the MachinePool webhook")
+
+		var mpOriginalReplicas int32
+		if hasMachinePool {
+			mpOriginalReplicas := *clusterResources.MachinePools[0].Spec.Replicas
```

Comment:
This creates a local scoped mpOriginalReplicas variable, is this intentional?

Reply:
nope, that was the bug why the test failed :-)